### PR TITLE
87 save customers

### PR DESCRIPTION
**Pull request by daros10**
Issue #87.

### CODE REVIEW

File: src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      name: ['', Validators.required],
+      description: [''],
+    });
+  }
+
+  ngOnInit() {
+    const messages$ = this.store.pipe(select(getStatusMessage));
+    this.saveSubscription = messages$.subscribe((valueMessage) => {
+      this.setStatusOnScreen(valueMessage);
+      console.log(valueMessage);
```
Comment (jorgecod) [inline src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.ts:32]: Please, delete this **console.log**

File: src/app/modules/customer-management/services/customer.service.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+})
+export class CustomerService {
+  baseUrl = `${environment.timeTrackerApiUrl}/customers`;
+
+  constructor(private http: HttpClient) {}
+
+  createCustomer(customerData): Observable<any> {
+    const body = {
+      ...customerData,
+      tenant_id: '4225ab1e-1033-4a5f-8650-0dd4950f38c8',
```
Comment (jorgecod) [inline src/app/modules/customer-management/services/customer.service.ts:17]: @enriquezrene  Do we still need to set a tenant_id fake?

File: src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
+import { Component, Input, Output, EventEmitter, OnDestroy, OnInit } from '@angular/core';
+import { Store, select } from '@ngrx/store';
+import { CustomerState, CreateCustomer } from 'src/app/modules/customer-management/store';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
```
Comment (jorgecod) [inline src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.ts:4]: Please sort imports by module.

File: src/app/modules/customer-management/services/customer.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import { TestBed, inject } from '@angular/core/testing';
+
+import { CustomerService } from './customer.service';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
```
Comment (jorgecod) [inline src/app/modules/customer-management/services/customer.service.spec.ts:4]: Please sort imports by module

File: src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.spec.ts [hunk 5]
```diff
@@ -0,0 +1,106 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { CreateCustomerComponent } from './create-customer';
+import { MockStore, provideMockStore } from '@ngrx/store/testing';
+import { CustomerState, CreateCustomer } from 'src/app/modules/customer-management/store';
+import { FormBuilder } from '@angular/forms';
+import { Customer } from 'src/app/modules/shared/models/customer.model';
```
Comment (jorgecod) [inline src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.spec.ts:7]: The customer.model must be added in the index file in **'src/app/modules/shared/models/index.ts'**

File: src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.spec.ts [hunk 6]
```diff
@@ -0,0 +1,106 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { CreateCustomerComponent } from './create-customer';
+import { MockStore, provideMockStore } from '@ngrx/store/testing';
+import { CustomerState, CreateCustomer } from 'src/app/modules/customer-management/store';
+import { FormBuilder } from '@angular/forms';
```
Comment (jorgecod) [inline src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.spec.ts:6]: Please sort imports by module

File: src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.html [hunk 7]
```diff
@@ -0,0 +1,40 @@
+<div class="container">
+  <form style="width: 600px;" [formGroup]="customerForm" (ngSubmit)="onSubmit(customerForm.value)">
+    <div class="form-group">
+      <div
+        *ngIf="showAlert && messageToShow !== ''"
+        [ngClass]="{'bg-secondary': messageToShow == 'Customer create successfully!', 'bg-primary': messageToShow != 'Customer create successfully!'}"
```
Comment (enriquezrene) [inline src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.html:6]: typo:
Customer **created** successfully!

File: src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.spec.ts [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+    spyOn(store, 'dispatch');
+
+    component.ngOnInit();
+
+    component.onSubmit(customerData);
+
+    component.setStatusOnScreen('Customer create successfully!');
+
+    expect(component.messageToShow).toEqual('Customer create successfully!');
```
Comment (enriquezrene) [inline src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.spec.ts:87]: Customer **created** successfully

File: src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.spec.ts [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+    expect(component.isActiveItemTabs).toBeTrue();
+  });
+
+  it('should be disabled tabs and show message Something went wrong creating customer! ', () => {
```
Comment (enriquezrene) [inline src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.spec.ts:91]: it('should disable tabs and show a message "An error occurred, try again later." '

File: src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.ts [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+import { getStatusMessage } from 'src/app/modules/customer-management/store/customer-management.selectors';
+
+@Component({
+  selector: 'app-create-customer',
+  templateUrl: './create-customer.html',
+  styleUrls: ['./create-customer.scss'],
+})
+export class CreateCustomerComponent implements OnInit, OnDestroy {
+  customerForm: FormGroup;
+  @Input() isActiveItemTabs: boolean;
```
Comment (enriquezrene) [inline src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.ts:15]: A better name would be:
areTabsActive

File: src/app/modules/customer-management/services/customer.service.ts (hunk 2 above)
Comment (enriquezrene) [reply src/app/modules/customer-management/services/customer.service.ts:17]: I think @daros10 needs to talk to @macrisguncay to understand how to get the tenant_id from the session_storage. Good catch @jorgecod, ty.

Comment (daros10) [reply src/app/modules/customer-management/services/customer.service.ts:18]: changes was applied except this.

File: src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.ts (hunk 1 above)
Comment (daros10) [reply src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.ts]: donde

File: src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.ts (hunk 3 above)
Comment (daros10) [reply src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.ts:2]: done

File: src/app/modules/customer-management/services/customer.service.spec.ts (hunk 4 above)
Comment (daros10) [reply src/app/modules/customer-management/services/customer.service.spec.ts]: done

File: src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.spec.ts (hunk 5 above)
Comment (daros10) [reply src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.spec.ts]: done

File: src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.spec.ts (hunk 6 above)
Comment (daros10) [reply src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.spec.ts]: done

File: src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.html (hunk 7 above)
Comment (daros10) [reply src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.html]: done

File: src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.spec.ts (hunk 8 above)
Comment (daros10) [reply src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.spec.ts]: done

File: src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.spec.ts (hunk 9 above)
Comment (daros10) [reply src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.spec.ts]: done

File: src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.ts (hunk 10 above)
Comment (daros10) [reply src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.ts]: done

File: src/app/modules/customer-management/services/customer.service.ts (hunk 2 above)
Comment (macrisguncay) [reply src/app/modules/customer-management/services/customer.service.ts]: @daros10 you can get the real tenant id and bearer token from session storage, I implemented these methods in AzureAdB2CService, please check it and use them please.

Comment (daros10) [reply src/app/modules/customer-management/services/customer.service.ts]: thank you @macrisguncay , @enriquezrene all review requesting was applied.

File: src/app/modules/customer-management/services/customer.service.ts [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+  baseUrl = `${environment.timeTrackerApiUrl}/customers`;
+
+  constructor(private http: HttpClient, private service: AzureAdB2CService) {}
+
+  createCustomer(customerData): Observable<any> {
+    const body = {
+      ...customerData,
+      tenant_id: this.service.getTenantId(),
+    };
+    console.log(this.service.getTenantId());
```
Comment (macrisguncay) [inline src/app/modules/customer-management/services/customer.service.ts:21]: remove the console.log please.

**Pull request merged**